packages/block-editor/src/components/writing-flow/use-input.js

```diff
@@ -59,6 +77,23 @@ export default function useInput() {
 			}
 
 			if ( ! hasMultiSelection() ) {
+				const { ownerDocument } = node;
+				if ( node === ownerDocument.activeElement ) {
+					if ( event.key === 'End' || event.key === 'Home' ) {
+						const selectionRoot = getSelectionRoot( ownerDocument );
+						const selection =
+							ownerDocument.defaultView.getSelection();
+						selection.selectAllChildren( selectionRoot );
+						const method =
+							event.key === 'End'
+								? 'collapseToEnd'
+								: 'collapseToStart';
+						selection[ method ]();
+						event.preventDefault();
+						return;
+					}
+				}
+
 				if ( event.keyCode === ENTER ) {
 					if ( event.shiftKey || __unstableIsFullySelected() ) {
```

Comment:
Not from here, but curious how `Enter` and `shift` are used..  Any chance you remember?

Reply:
Yes, this is Enter behaviour for partial selection. So select two paragraphs partially and press Enter, this handles split for them.